compare_template_text/normalize_template_text.py

```diff
@@ -0,0 +1,65 @@
+import re
+import sys
+
+"""This class normalizes the template by implementing the Omitable and Replaceable text 
+normalizations. """
+
+class NormalizeTemplate:
+    
+    def __init__(self,text_string,template_string):
+        self.text_string = text_string
+        self.template_string = template_string
+                
+    def normalize_template(self):
+        self.remove_bulletted_text()
+        self.remove_omitable_text()
+        self.remove_repeating_chars()
+    
+    def return_normalized_template(self):
+        
+        """ Returns the Normalized template after normalization techniques. """
+        print(self.template_string)
+        return self.template_string
+    
+    def return_normalized_text(self):
+        """ Returns the normalized text after normalization techniques. """
+        
+        print(self.text_string)
+        return self.text_string
+    
+    def remove_omitable_text(self):
+        """ Removes text from the template and the text that is optional. So, the text 
+        which is optional and present matches and gets replaced by '' while the text that doesn't match will 
+        remain as it is thus giving an unmatch. """
+         
+        lists = re.findall('(?<=\<\<beginoptional\>\>).*?(?=\<\<endoptional\>\>)',
+                           self.template_string)
+        # print(lists)
+        for x in lists:
+            if(x.startswith('*')):
+                x = x[1:]
+
+            if(x.endswith('*')):
+                x = x[:-1]
+
+            self.text_string = self.text_string.replace(x,'')
+            self.template_string = self.template_string.replace(x,'')
+        # print(lists)
+        return
+    
+    def remove_bulletted_text(self):
```

Comment:
If you plan to handle general `<var` text matching, this may not be needed

Reply:
@goneall Yes I tried that but it isn't possible in regex. Actually in order to match the text in between 2 tags we need a fixed text before and after the ending ">>". So a generalized 'var' won't work and I will have to specify whether it's a copyright or a bullet.

Reply:
@anshuldutt21 do you plan to support regular `<<var ...>>` expressions in a future PR?  If so, we can discuss the approach then.

Reply:
@goneall is it really needed to generalize it? There are just 2 cases of bulleted text and copyright replaceable text which I have dealt with separately. The regex won't allow us to generalize it. I would have to use a different approach for that.